public/omni/infrastructure-and-extensions/writing-infrastructure-providers.mdx

```diff
@@ -0,0 +1,256 @@
+---
+title: How to Write an Infrastructure Provider
+---
+
+## Example Provider
+
+Reference implementation is our [KubeVirt provider](https://github.com/siderolabs/omni-infra-provider-kubevirt/).
+The easiest way is to copy it and change the code that interacts with the platform.
+
+## Overview
+
+This guide explains how Omni dynamically provisions machines through infrastructure providers and how to implement one.
+
+If you already have a `MachineClass` that provisions machines dynamically, the following flow outlines how Omni interacts with it:
+
+1. Create a `MachineSet` that uses a class with autoprovisioning enabled.
+2. Omni creates a `MachineRequestSet` with the same name as the `MachineSet`, aligning the desired number of machines.
```

Comment:
we should probably externally link to this. i am not confident that every single person that lands on this page, will know how to create these and what these are